lonestardist/tc doesn't build

```
[  9%] Building CUDA object lonestardist/tc/CMakeFiles/tc_cuda.dir/tc_cuda.cu.o
/home/ddn/w/katana/lonestardist/tc/tc_cuda.cu:9:10: fatal error: kernels/reduce.cuh: No such file or directory
 #include "kernels/reduce.cuh"
          ^~~~~~~~~~~~~~~~~~~~
```

--- a/lonestardist/CMakeLists.txt
+++ b/lonestardist/CMakeLists.txt
@@ -167,6 +167,12 @@ function(add_test_dist app input)
     endif()
   endforeach()
 endfunction()
+
+# currently tc only has a GPU implementation
+if(ENABLE_HETERO_GALOIS)
+  add_subdirectory(tc)
+endif()
+
 add_subdirectory(bc)
 add_subdirectory(bfs)
 add_subdirectory(cc)
@@ -178,8 +184,3 @@ add_subdirectory(sssp)
 
 add_subdirectory(partition)
 
-# currently tc only has a GPU implementation
-if(ENABLE_HETERO_GALOIS)
-  # tc does not currently build
-  add_subdirectory(tc EXCLUDE_FROM_ALL)
-endif()

--- a/lonestardist/tc/tc_cuda.cu
+++ b/lonestardist/tc/tc_cuda.cu
@@ -10,8 +10,7 @@ void kernel_sizing(CSRGraph &, dim3 &, dim3 &);
 #include "tc_cuda.cuh"
 #include "moderngpu/kernel_segsort.hxx"
 #include <cuda_profiler_api.h>
-using namespace mgpu;
-standard_context_t context;
+mgpu::standard_context_t context;
 #define WARP_SIZE 32
 
 inline __device__ unsigned long intersect(CSRGraph graph, index_type u, index_type v) {
@@ -121,7 +120,7 @@ __global__ void warp(CSRGraph graph, unsigned begin, unsigned end, HGAccumulator
 }
 
 void sort_cuda(struct CUDA_Context* ctx) {
-        segmented_sort(ctx->gg.edge_dst, ctx->gg.nedges, (const int *) ctx->gg.row_start + 1, ctx->gg.nnodes - 1, less_t<int>(), context);
+        mgpu::segmented_sort(ctx->gg.edge_dst, ctx->gg.nedges, (const int *) ctx->gg.row_start + 1, ctx->gg.nnodes - 1, mgpu::less_t<int>(), context);
 }
 
 void TC_cuda(unsigned __begin, unsigned __end, unsigned long & num_local_triangles, struct CUDA_Context* ctx) {

--- a/lonestardist/tc/tc_cuda.cu
+++ b/lonestardist/tc/tc_cuda.cu
@@ -6,13 +6,12 @@
 
 void kernel_sizing(CSRGraph &, dim3 &, dim3 &);
 #define TB_SIZE 256
-#include "kernels/reduce.cuh"
+#include "moderngpu/kernel_reduce.hxx"
 #include "tc_cuda.cuh"
-#include "kernels/segmentedsort.cuh"
-#include "moderngpu.cuh"
-#include "util/mgpucontext.h"
+#include "moderngpu/kernel_segsort.hxx"
 #include <cuda_profiler_api.h>
-mgpu::ContextPtr mgc;
+using namespace mgpu;
+standard_context_t context;
 #define WARP_SIZE 32
 
 inline __device__ unsigned long intersect(CSRGraph graph, index_type u, index_type v) {
@@ -83,7 +82,6 @@ __global__ void warp(CSRGraph graph, unsigned begin, unsigned end, HGAccumulator
 	unsigned thread_id   = blockIdx.x * blockDim.x + threadIdx.x;
 	unsigned thread_lane = threadIdx.x & (WARP_SIZE-1);            // thread index within the warp
 	unsigned warp_id     = thread_id   / WARP_SIZE;                // global warp index
-	unsigned warp_lane   = threadIdx.x / WARP_SIZE;                // warp index within the CTA
 	unsigned num_warps   = (TB_SIZE / WARP_SIZE) * gridDim.x;   // total number of active warps
 
 	__shared__ cub::BlockReduce<unsigned long, TB_SIZE>::TempStorage num_local_triangles_ts;
@@ -123,8 +121,7 @@ __global__ void warp(CSRGraph graph, unsigned begin, unsigned end, HGAccumulator
 }
 
 void sort_cuda(struct CUDA_Context* ctx) {
-	mgc = mgpu::CreateCudaDevice(ctx->device);
-	mgpu::SegSortKeysFromIndices(ctx->gg.edge_dst, ctx->gg.nedges, (const int *) ctx->gg.row_start + 1, ctx->gg.nnodes - 1, *mgc);
+        segmented_sort(ctx->gg.edge_dst, ctx->gg.nedges, (const int *) ctx->gg.row_start + 1, ctx->gg.nnodes - 1, less_t<int>(), context);
 }
 
 void TC_cuda(unsigned __begin, unsigned __end, unsigned long & num_local_triangles, struct CUDA_Context* ctx) {

